ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeTaskItem.java

```diff
@@ -192,15 +209,37 @@ public boolean canExecute(Supplier<Integer> maxRetry) {
     if (getOptimizeStatus() == OptimizeStatus.Init) {
       return true;
     } else if (getOptimizeStatus() == OptimizeStatus.Failed) {
+      // clear useless files produced by failed full optimize task support hive
+      String location = getOptimizeTask().getProperties().get(OptimizeTaskProperties.CUSTOM_HIVE_SUB_DIRECTORY);
+      if (location != null) {
+        try {
+          ArcticTable arcticTable = ServiceContainer.getOptimizeService()
+              .getTableOptimizeItem(getTableIdentifier()).getArcticTable();
+          for (FileStatus fileStatus : arcticTable.io().list(location)) {
+            String fileLocation = fileStatus.getPath().toUri().getPath();
+            if (fileLocation.contains(optimizeRuntime.getAttemptId())) {
```

Comment:
What is the content of attempt id ?
Is it safe to find out files produced by a certain task by this operation?

Reply:
The attempt id produced by Math.abs(ThreadLocalRandom.current().nextInt())

Reply:
I will use a regular expression to match the file name

Reply:
Random integers seem to collide easily, where did this attempt id generate from? executor or AMS?

You can comment the file name pattern here to help others understand your code.

Reply:
This attempt id is generated by the executor.